best-practices/db_reset.md

```diff
@@ -0,0 +1,36 @@
+# Resetting the db for a deployed Rails application
+
+Unfortunately:
+* A simple `rails db:reset` will not work to reset the db for a deployed application.
+* Servers are not consistent, so no single process will work in all cases. 
+
+However, the following should handle most cases:
+1. Stop all services connecting to the db. This can be accomplished by executing the following as ksu on each of the application's servers:
+```
+systemctl stop apache2
+systemctl stop passenger
+systemctl stop sneakers
+systemctl stop sidekiq-*
+```
+Note: An application may only have some of these services. There is no harm in executing the command if a service is absent.
+
+2. Drop the database: `RAILS_ENV=production DISABLE_DATABASE_ENVIRONMENT_CHECK=1 bundle exec rails db:drop`
```

Comment:
i would lean ever so slightly towards not including `DISABLE_DATABASE_ENVIRONMENT_CHECK=1` in the command to cut and paste, requiring the user to type it in separately, just out of an abundance of caution to force one to basically click "yes i really want to do this" since we all often have many terminal windows open, sometimes to a mix of prod and stage.  but also we're good about DB backups, and confirmation prompts aren't foolproof, so wouldn't block over this.

Reply:
(also thanks for the quick and thorough writeup 🙂)

Reply:
Good idea. Done.